### PR TITLE
add object.assign polyfill and skip tests ctx

### PR DESCRIPTION
**Pull request by e111077**
Tests will not run at all on IE even if you want to skip the tests without the Polyfill. Additionally, I added functionality to skip tests. Useful when your storage uses features not available in browsers that `app-storage` supports.

### CODE REVIEW

**Comment by e111077** [comment]
Ignore failing tests. Tests never worked and I recently turned them on for the 2.0 polyfill

File: test/app-storage-compatibility-suite.html [hunk 1]
```diff
@@ -7,10 +7,36 @@
 Code distributed by Google as part of the polymer project is also
 subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
 -->
+
 <script>
 (function() {
   'use strict';
 
+  if (typeof Object.assign != 'function') {
```
Comment (valdrinkoshi) [inline test/app-storage-compatibility-suite.html]: Please add the reference to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#Polyfill

File: test/app-storage-compatibility-suite.html [hunk 2]
```diff
@@ -36,6 +63,12 @@
     Object.assign(context, extendedContext);
```
Comment (valdrinkoshi) [inline test/app-storage-compatibility-suite.html]: Any reason not to remove `Object.assign` and instead do this?
```js
if (extendedContext) {
  for (var key in extendedContext) {
    context[key] = extendedContext[key];
  }
}

Comment (e111077) [inline test/app-storage-compatibility-suite.html]: Agreed

Comment (e111077) [inline test/app-storage-compatibility-suite.html]: done.

**Comment by valdrinkoshi** [comment]
LGTM, good for merge 
@e111077 I think this should be merged in `2.0-preview` branch only, right?

**Pull request merged**